src/snowflake/snowpark/modin/plugin/_internal/io_utils.py

```diff
@@ -126,36 +129,37 @@ def get_columns_to_keep_for_usecols(
     ValueError
         If column(s) expected in `usecols` are not found in frame's columns `columns`.
     """
-
-    if not isinstance(usecols, list):
-        keep = [column for column in columns if usecols(column)]
+    _usecols = usecols
+    if callable(_usecols):
+        keep = [column for column in columns if _usecols(column)]
+    elif len(_usecols) <= 0:
```

Comment:
```suggestion
    elif len(_usecols) == 0:
```

nit: this looks weird to me :)

Reply:
This is my programming paranoia ;) changed.